### PR TITLE
feat(deque): Implement append and join methods with comprehensive tests

### PR DESCRIPTION
**Pull request by PingGuoMiaoMiao**
feat(deque): Add append method with tests

- Implement O(1) amortized append() for deque
- Add 9 test cases covering edge scenarios
- Document behavior in method signature

feat(deque): Optimize join method

- Replace string concatenation with Buffer
- Add type-safe generic support
- Match existing test patterns

### CODE REVIEW

**Comment by peter-jerry-ye** [comment]
`join` is implemented in #2191

File: deque/deque.mbt [hunk 1]
```diff
@@ -1297,3 +1297,45 @@ pub impl[A : @json.FromJson] @json.FromJson for T[A] with from_json(json, path)
...
+///
+/// Example:
+/// ```moonbit
+/// let dq = @deque.of([1, 2, 3])
+/// dq.append(4)
+/// inspect(dq, content="@deque.of([1, 2, 3, 4])")
+/// ```
+pub fn[A] append(
+  self : T[A], // The deque to modify
+  elem : A // The element to append
```
Comment (peter-jerry-ye) [inline deque/deque.mbt]: Appending only one element is not consistent with `Array::append`.

**Comment by PingGuoMiaoMiao** [comment]
> `join`[在#2191](https://github.com/moonbitlang/core/pull/2191)中实现[](https://github.com/moonbitlang/core/pull/2191)

I have deleted the join I implemented in pr

File: deque/deque.mbt [hunk 2]
```diff
@@ -1283,3 +1283,49 @@ pub impl[A : @json.FromJson] @json.FromJson for T[A] with from_json(json, path)
...
+///   * `self` : The deque to modify
+///   * `other` : The array to append (must be Array[T])
+///
+/// Example:
+/// ```moonbit
+/// let dq = @deque.of([1, 2])
+/// dq.append([3, 4])  
+/// inspect(dq, content="@deque.of([1, 2, 3, 4])")
+/// ```
+pub fn[A] append(self : T[A], other : Array[A]) -> Unit {
```
Comment (peter-jerry-ye) [inline deque/deque.mbt:1301]: `other` should be another `T[A]`.

File: deque/deque.mbt [hunk 3]
```diff
@@ -1283,3 +1283,49 @@ pub impl[A : @json.FromJson] @json.FromJson for T[A] with from_json(json, path)
...
+  } else {
+    let first_chunk_len = buf_cap - tail_pos
+    for i = 0; i < first_chunk_len; i = i + 1 {
+      self.buf[tail_pos + i] = other[i]
+    }
+    for i = 0; i < add_len - first_chunk_len; i = i + 1 {
+      self.buf[i] = other[first_chunk_len + i]
+    }
+    self.tail = add_len - first_chunk_len - 1
+  }
```
Comment (peter-jerry-ye) [inline deque/deque.mbt:1329]: Should use `blit`. Please wait until the other PR is merged, or you can rebase onto that branch.
